awscli/customizations/wizard/selectmenu.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from prompt_toolkit import Application
+from prompt_toolkit.utils import get_cwidth
+from prompt_toolkit.layout import Layout, FloatContainer, Float
+from prompt_toolkit.layout.controls import UIControl, UIContent
+from prompt_toolkit.layout.screen import Point
+from prompt_toolkit.layout.dimension import Dimension
+from prompt_toolkit.key_binding.key_bindings import KeyBindings
+from prompt_toolkit.layout.margins import ScrollbarMargin
+from prompt_toolkit.layout.containers import ScrollOffsets, Window
+
+
+def select_menu(items, display_format=None, max_height=10):
```

Comment:
It would also be nice if this was searchable via typing in a search query. It is pretty tedious when you can only scroll if you have a lot of options. How difficult would that be to add?

Reply:
It's not really the intent of this widget to allow you to type something, it's specifically designed so you can't.

The way we were planning on handling this is at the UI layer that uses these widgets.  Basically the UI would be smart enough that if the number of choices is over a certain threshold it would switch to the default prompt-toolkit prompt that allows you to type.

Reply:
Sounds fine to me